src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -111,48 +123,100 @@ static CallingConvention getCallingConvention(FunctionPointerCallingConventionSy
+                        //
+                        //  * We prepend the identifier with the string CallConv
+                        //  * We look only at types defined in the System.Runtime.CompilerServices namespace.
+                        //  * We look only at types defined in the core library of the application, which is the library that defines
+                        //    System.Object and has no dependencies.
+                        //
+                        // If lookup succeeds on all of the identifiers specified in an unmanaged_calling_convention, we encode the
+                        // CallKind as unmanaged, and encode each of the resolved types in the set of modopts at the beginning of
+                        // the function pointer signature.
+
                         switch (callingConventionSyntax.UnmanagedCallingConventionList)
                         {
                             case null:
+                                checkUnmanagedSupport(typeBinder, callingConventionSyntax.ManagedOrUnmanagedKeyword.GetLocation(), diagnostics, isExtensionError: false);
                                 return CallingConvention.Unmanaged;
 
                             case { CallingConventions: { Count: 1 } specifiers }:
                                 return specifiers[0].Name switch
                                 {
+                                    // Special identifiers cases
                                     { ValueText: "Cdecl" } => CallingConvention.CDecl,
                                     { ValueText: "Stdcall" } => CallingConvention.Standard,
                                     { ValueText: "Thiscall" } => CallingConvention.ThisCall,
                                     { ValueText: "Fastcall" } => CallingConvention.FastCall,
-                                    // PROTOTYPE(func-ptr): Handle unrecognized specifiers
-                                    var name => reportBadConventionAndReturn(name.ValueText, name, diagnostics)
+
+                                    // Unknown identifier case
+                                    _ => handleSingleConvention(specifiers[0], typeBinder, customModifiers, diagnostics)
                                 };
 
                             case { CallingConventions: { Count: 0 } } unmanagedList:
                                 // Should never be possible from parser-constructed code (parser will always provide at least a missing identifier token),
                                 // so diagnostic quality isn't hugely important
-                                return reportBadConventionAndReturn(givenConvention: "", unmanagedList, diagnostics);
+                                if (!unmanagedList.ContainsDiagnostics)
+                                {
+                                    diagnostics.Add(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, unmanagedList.OpenBracketToken.GetLocation(), "");
```

Comment:
>ERR_InvalidFunctionPointerCallingConvention [](start = 62, length = 43)

Is this message going to be confusing?

Reply:
Maybe, but as the comment says you can't observe this from parser-created code so I don't think diagnostic quality is particularly important.

---
In reply to: [457734288](https://github.com/dotnet/roslyn/pull/46110#discussion_r457734288) [](ancestors = 457734288)